Lib/enum.py

```diff
@@ -250,7 +250,11 @@ def __new__(metacls, cls, bases, classdict):
 
         # double check that repr and friends are not the mixin's or various
         # things break (such as pickle)
+        # however, if the method is defined in the Enum itself, don't replace
+        # it
         for name in ('__repr__', '__str__', '__format__', '__reduce_ex__'):
+            if name in classdict:
```

Comment:
It works only if `__str__ = object.__str__` in the definition of the Enum subclass.

I think we should check whether `__str__` was inherited from `object` or from other class. E.g.:
```python
class A(int):
    __str__ = object.__str__

class B(A, Enum):
    pass

assert B.__str__ is object.__str__
```

Reply:
In other words, we should check whether the class where the method was defined is enumerate before `Enum` in the `__mro__` order.

```
>>> B.__mro__
(<enum 'B'>, <class '__main__.A'>, <class 'int'>, <enum 'Enum'>, <class 'object'>)
```

Reply:
Those four methods must be from an Enum or a non-data-type mix-in [1]; if the first one found via inheritance is from a data type, then the method will be copied into the being-constructed enum from the base Enum [2].  In your example above, the assert will, and must, fail because class `A` is a data type.  

[1] a data type is considered to be anything with a non-`object' '__new__` method

[2] the base Enum is the one listed *last* in the class header; so in `class MyEnum(Enum1, Enum2, IntEnum):` the base Enum would be `IntEnum`